### ISSUE TITLE
Expanding the contents of the details panel when viewing a page

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#48482]
When you navigate to a Page in the Site Editor here's what you see:

<img width="1149" alt="Screenshot 2023-02-27 at 13 25 19" src="https://user-images.githubusercontent.com/846565/221575553-fd93b8e5-5e61-42d2-a62e-8e32a3c42282.png">

In the future it will be nice to have a UI for making high-level page edits in the dark sidebar. In the interim it would be good to at least include the page URL (most likely as an actual link):

<img width="359" alt="Screenshot 2023-02-27 at 13 30 07" src="https://user-images.githubusercontent.com/846565/221576430-1de18385-e31d-465e-b597-952f9058fdb7.png">

We should also tidy up the spacing / alignment, but it's probably okay for that to be a separate effort.

**Comment by richtabor** [comment]
> (most likely as an actual link)

Are you thinking an ExternalLink?

**Comment by jeryj** [comment]
Update: I misunderstood which area of the site editor we were talking about. I thought it was Site Editor -> Templates -> Page template, not Site Editor -> Navigation -> Page List -> Individual Page.

----------

I'm feeling stuck on this issue, as I can't find an easy way to identify which page is assigned to the template. 

**tl;dr:** Templates don't care about what posts/pages/etc are using them, so there isn't a way to easily access which page is referenced from this template. And without knowing which page is being used in the template, we don't know which link to use. It would be technically possible to find it, but in the current architecture, this would be a very expensive operation. To do this efficiently, it would take a change in core, and likely not a small one.

Also, there's a very real chance that I'm wrong here and someone will chime in with "Oh, just use X function."

Sorry for the long post — I wanted to document my research since it did take awhile do debug this and it was too much to keep in my head all at once 😅 

### How I investigated this since this is my first time looking at templates

I started at the template edit screen javascript to see if there was a reference to the page that is using the template. I had thought since the page title was referenced in the template title (for example, in this issue screenshot, "Sample Page"), it would mean it was referencing the Sample Page itself. And if we have access to the page title, we probably have access to the page. Easy!

However, it is getting the "Sample Page" title from the template title in the database, not the original page. When a custom template is created for a page, it sets the title for that template as the page it came from so they'll match. Similarly, if you update the original post/page that created the template, it will also update the custom template name to match the updated name. This is set in the database for the template custom post type, not by referencing the page. So, the "Sample Page" in this issue screenshot _isn't_ the Sample Page title, it's the template title named "Sample Page."

_Note: I still need to debug how the template title is kept up-to-date with the page title, likely using the `save_post` or like hook. I'm curious if there's some faster way of identifying the template by page that I'm unaware of._

Since I wasn't making progress on how to identify a page by the template, I tried figuring out how a page identifies its template. If it was quick enough, maybe we could just search all pages and return the template. For example, I see that there's a request already returning some info about the pages in this editor screen. If that meta info included or could include the template ID it was using, then we could quickly locate it without making additional requests.

However, this would be quite an expensive query with the current architecture as there is no stored relationship between a page and its custom template. 

When a page is being edited, to identify the template it uses, it runs through [`resolve_block_template()`](https://developer.wordpress.org/reference/functions/resolve_block_template/), which looks through all possible templates for this page and return the one with the highest priority match. We'd have to do this for every page, checking every template until we had our match, which would be quite expensive.

### Why I don't think we should do a change in core to address this
So, the issue is we don't have a quick reference between the page and template IDs. I had thought maybe we could store some meta info for the template, like, this template will always use x page, so store the `post_id` in the `post_meta` table for that template to make this faster to lookup. But I think this would be ill-advised since templates work by priority not by a direct association. 

Templates work by viewing all possible sources a template could come from and be valid for that page/post, then selecting the one with the highest priority. For example, an about page could use a file within the theme named:
- `page.php`
- `about-page.php`
- `page-${post_id}.php`
- or the custom template post-type (what's being used for the templates in this issue). 

And if it's a child theme, it also checks these same things within the parent theme. So, it was never built to have a 1:1 relationship of a template to a page, but in our custom templates, it forces a 1:1 relationship in some instances by being more and more specific about the theme (this custom template post type will have the highest priority when checking for which template to apply). But it still has to go through all the steps to locate the page (get the page info, sort through all potential templates, reorder them by best match, return the best match).

To do all of this to access the page related to the custom post here would be quite expensive, so I think it's best to close this issue unless it was the start of a large overhaul of the template editor or blocking a future design direction that needs to get solved. If this issue was intended to be an easy win, unfortunately, it isn't at the moment unless I'm missing something (very possible). 😕

**Comment by jameskoster** [comment]
> I'm feeling stuck on this issue, as I can't find an easy way to identify which page is assigned to the template.

Forgive a potentially silly question, why do we _need_ to identify that assignation here? This issue is purely about querying the relevant page and displaying the URL.

**Comment by jeryj** [comment]
> Forgive a potentially silly question, why do we need to identify that assignation here? 

No worries! I thought this issue would be much simpler than it is, so I'm still holding out for something obvious I was overlooking to come up, so please keep asking! 

> This issue is purely about querying the relevant page and displaying the URL.

Agreed! Unfortunately, that is what's difficult to do. There isn't a non-expensive way of identifying the page and URL from only having the template info.

**Comment by scruffian** [comment]
Here's an idea: https://github.com/WordPress/gutenberg/pull/48650
